### ISSUE TITLE
Methods involving variants not exposed to script (JS/TS and C#)

### ISSUE THREAD

**Issue by mattbenic** [AtomicGameEngine/AtomicGameEngine#995]
eg Material.SetShaderParameter

**Comment by JoshEngebretson** [comment]
Closing this via #1275, I want to avoid introducing a bunch of Variant style API (and there isn't a ton of it).  So, we can hand bind variant setters/getters we need.  I added Material.SetShaderParameter, if there is other API which uses Variant needed, let's reopen this with list of methods.

**Comment by raheelx** [comment]
hi josh :) just out of curiosity, why do want to avoid introducing variant style API?

**Comment by JoshEngebretson** [comment]
Hi @raheelx , C# currently uses the generic RefCounted [ScriptVariant](https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Source/Atomic/Script/ScriptVariant.h) which is a convenience class making variants accessible using standard script binding generation (JS/TS/C#).  

If we find ourselves needing to marshal Variants a lot in C#, we should switch to a blittable Variant struct.  VariantMaps are also something that could steer this, as we use them for event data, the blittable Variant would be good in this case as well.

So, basically, don't want to add a bunch of Variant API which rides on ScriptVariant and instead figure out how much API surface we need with variants and of those methods which would be better constructed by flattening the valid data types, for example on Material.SetShaderParameter only a few data types are accepted "Color, Vector4, etc" and it would be better to simply have overloads for these types for clarity and performance than a Variant abstraction.
